### DISCUSSION TITLE
Why code chunk only shows half of the code?

### DISCUSSION THREAD

**Discussion by lizhiwei1994** [quarto-dev/quarto-cli#2203] [Q&A] [answered]
Hi everyone!

# Content
I am using quaro to make slides based on revealjs. and I set `width:1920` and `height:1080` in `YAML`.
# Question
I do not know why code chunk only shows half of the code.
Is there any way to make the displayed code as high as the image on the right.

<details>
  <summary>Half code in code chunk</summary>

  ![half_code](https://user-images.githubusercontent.com/59855957/187328105-a11bc82e-63bf-4872-b4b0-6f22e8a19add.png)
</details>

# Reproducible code
<details>
  <summary>Click me</summary>
  
~~~
---
title: "Untitled"
format: 
  revealjs:
    width: 1920
    height: 1080
editor: visual
---
```{r}
#| include: false
library(forestplot)

fp <- structure(list(V1 = c("Disease", "Arrhythmia", "Age", "    <65", 
"    >=65", "Sex", "    Male", "    Female", "", "Cerebrovascular", 
"Age", "    <65", "    >=65", "Sex", "    Male", "    Female", 
"", "CHD", "Age", "    <65", "    >=65", "Sex", "    Male", "    Female", 
"", "Heart failure", "Age", "    <65", "    >=65", "Sex", "    Male", 
"    Female"), V2 = c("Percentage Changes (95% CI)", "", "", 
"-2.14(-4.02 to -0.27)", "-0.31(-1.8 to 1.19)", "", "-1.22(-3.1 to 0.65)", 
"-0.55(-2.23 to 1.12)", "", "", "", "-0.11(-0.62 to 0.4)", "-0.05(-0.51 to 0.41)", 
"", "0.12(-0.36 to 0.59)", "-0.3(-0.86 to 0.25)", "", "", "", 
"1.67(0.68 to 2.67)", "0.84(0.09 to 1.58)", "", "1.28(0.42 to 2.15)", 
"1.01(0.18 to 1.83)", "", "", "", "1.23(-2.65 to 5.11)", "-0.67(-2.65 to 1.32)", 
"", "0.54(-1.95 to 3.03)", "-1.09(-3.6 to 1.42)"), V3 = c(NA, 
NA, NA, -2.14, -0.31, NA, -1.22, -0.55, NA, NA, NA, -0.11, -0.05, 
NA, 0.12, -0.3, NA, NA, NA, 1.67, 0.84, NA, 1.28, 1.01, NA, NA, 
NA, 1.23, -0.67, NA, 0.54, -1.09), V4 = c(NA, NA, NA, -4.02, 
-1.8, NA, -3.1, -2.23, NA, NA, NA, -0.62, -0.51, NA, -0.36, -0.86, 
NA, NA, NA, 0.68, 0.09, NA, 0.42, 0.18, NA, NA, NA, -2.65, -2.65, 
NA, -1.95, -3.6), V5 = c(NA, NA, NA, -0.27, 1.19, NA, 0.65, 1.12, 
NA, NA, NA, 0.4, 0.41, NA, 0.59, 0.25, NA, NA, NA, 2.67, 1.58, 
NA, 2.15, 1.83, NA, NA, NA, 5.11, 1.32, NA, 3.03, 1.42)), class = "data.frame", row.names = c(NA, 
-32L))
```

```{r fp1, fig.height=11}
#| output-location: column
#| echo: true
forestplot(
          labeltext = as.matrix(fp[,1:2]),
          mean = fp$V3, 
          lower = fp$V4,
          upper = fp$V5,
          is.summary = c(T,rep(c(T,F,F,F,F,F,F,F),3),T,F,F,F,F,F,F), 
          zero = 0, 
          boxsize = 0.5,  
          lineheight = unit(8,'mm'), 
          colgap = unit(2,'mm'), 
          lwd.zero = 1,
          lwd.ci = 1,
          col = fpColors(box='blue', 
                         summary='#8B008B', 
                         lines = 'black', 
                         zero = 'skyblue'), 
          xlab = 'The estimate', 
          lwd.xaxis = 1, 
          lty.ci = 1,
          ci.vertices = T, 
          ci.vertices.height = 0.15, 
          graph.pos = 2,
          fn.ci_norm = fpDrawNormalCI, 
          txt_gp = fpTxtGp(
                            ticks = gpar(cex = 1), 
                            xlab = gpar(cex = 1),  
                            label = gpar(cex = 1)),  
          hrzl_lines=list(  "1"  = gpar(lty=1,lwd=1, col="black",fill="black"), 
                            "2"  = gpar(lty=1,lwd=1, col="black",fill="black"),
                            "33" = gpar(lty=1,lwd=1, col="black",fill="black"),
                            "5"  = gpar(lty=1,lwd=60,lineend="butt",columns=c(2:3),col="#89898855"),
                            "8"  = gpar(lty=1,lwd=60,lineend="butt",columns=c(2:3),col="#89898855"),
                            "13" = gpar(lty=1,lwd=60,lineend="butt",columns=c(2:3),col="#89898855"),
                            "16" = gpar(lty=1,lwd=60,lineend="butt",columns=c(2:3),col="#89898855"),
                            "21" = gpar(lty=1,lwd=60,lineend="butt",columns=c(2:3),col="#89898855"),
                            "24" = gpar(lty=1,lwd=60,lineend="butt",columns=c(2:3),col="#89898855"),
                            "29" = gpar(lty=1,lwd=60,lineend="butt",columns=c(2:3),col="#89898855"),
                            "32" = gpar(lty=1,lwd=60,lineend="butt",columns=c(2:3),col="#89898855")
                            )
       
)

```

~~~
</details>

**Comment by cderv** [comment, marked as answer]
We have a `max-height` set on the code block for reveal.js presentation 

https://github.com/quarto-dev/quarto-cli/blob/cb578b9753862fdf22a4fa3a46ad213b7de596b5/src/resources/formats/revealjs/quarto.scss#L247

which already increase the **revealjs** default one from the template that we include
https://github.com/quarto-dev/quarto-cli/blob/f29aacff33fc8f49174df17c161998811b7374b0/src/resources/formats/revealjs/reveal/css/theme/template/theme.scss#L195
(source of template in https://github.com/hakimel/reveal.js/blob/1be851658f0eefd7fca2fad87b543c874eb322c0/css/theme/template/theme.scss#L195)

So that is how reveal.js works with code block by default it seems, and we already made it bigger. 

We could try improve the default, but in some other situation with text paragraph under code blocks for example, the code block would take the whole slide and text probably would not be visible. I believe this is the reason why this is set that way. 

Not that you can overwrite the theme: https://quarto.org/docs/presentations/revealjs/themes.html#customizing-themes 
and change `max-width` for this. 

To make that automatic, we would need to find a way to compute the available space in the column and change the width accordingly 🤔

**Reply by cderv** [reply]
Just a notice that we will probably make that available as an option in YAML : https://github.com/quarto-dev/quarto-cli/issues/1943

**Reply by mcanouil** [reply]
Why not set `max-height: 90%`? 🤔
For now customisation can be achieved with knitr engine (and other engine in the future #2040) using `class-source` option for example.

**Reply by cderv** [reply]
> Why not set max-height: 90%? 🤔

yes why not. We need to find the good solution to tweak reveal.js and cover our different features.
